drivers/spi/spi_bitbang.c

```diff
@@ -45,13 +45,19 @@ static int spi_bitbang_configure(const struct spi_bitbang_config *info,
 
 	const int bits = SPI_WORD_SIZE_GET(config->operation);
 
-	if (bits > 16) {
-		LOG_ERR("Word sizes > 16 bits not supported");
+	if (bits > 32) {
+		LOG_ERR("Word sizes > 32 bits not supported");
 		return -ENOTSUP;
 	}
 
 	data->bits = bits;
 	data->dfs = ((data->bits - 1) / 8) + 1;
+
+	/* As there is no uint24_t, it is assumed uint32_t will be used as the buffer base type. */
+	if (data->dfs == 3) {
+		data->dfs = 4;
```

Comment:
that does not sound right, see line 233: you will advance into tx/rx buffers by 4 even for dfs of 3. 

So far it seems you can remove that changes, as you anyway coalesce case 3 into case 4 in subsequent switches.

Reply:
```see line 233: you will advance into tx/rx buffers by 4 even for dfs of 3```
Yes, this is intended. Removing that line would force users to use uint24_t (which doesn't exist in zephyr, so probably uint8_t[3] would be required as a base type for the buffer). Not sure this is best idea, as for some platforms it could cause issues regarding misalignment.

If you wish, I can change it, but I believe some hardware implementations accepts 4 bytes input, even when the `dfs` is 3, to avoid the misalignment mentioned above.